File: Section1 - Basics/Patterns/pattern7.js
// BruteForce O(n^2)
function printTriangle(n) {
	let str = "";
	for (let i = 0; i < n; i++) {
		for (let j = n; j > i + 1; j--) {
			str += " ";
		}
		for (let k = 0; k <= i; k++) {
			str += "*";
		}
		for (let l = 0; l < i; l++) {
			str += "*";
		}
		str += "\n";
	}
	console.log(str);
}

printTriangle(4);

// Optimised O(n^2)
function printTriangle(n) {
	let str = "";
	for (let i = 1; i <= n; i++) {
		for (let j = 1; j <= n - i; j++) {
			str += " ";
		}
		for (let k = 1; k <= 2 * i - 1; k++) {
			str += "*";
		}
		str += "\n";
	}
	console.log(str);
}

printTriangle(4);

File: Section1 - Basics/Patterns/pattern7.java
// BruteForce O(n^2)
void printTriangle(int n) {
    // code here
    for(int i=0; i<n; i++) {
        for(int j=n; j>i+1; j--) {
            System.out.print(" ");
        }
        for(int k=0; k<=i; k++) {
            System.out.print("*");
        }
        for(int j=0; j<i; j++) {
            System.out.print("*");
        }
        System.out.print("\n");
    }
}

// Optimised O(n^2)
void printTriangle(int n) {
    // code here
    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n - i; j++) {
            System.out.print(" ");
        }
        for (int k = 1; k <= 2 * i - 1; k++) {
            System.out.print("*");
        }
        System.out.println();
    }
}